src/helpers/WalkHelper.php

```diff
@@ -54,7 +54,14 @@ public static function craftyArrayWalk(&$array, $callable, $userdata = null)
 
 		if (is_callable($callable))
 		{
-			return array_walk($array, $callable, $userdata);
```

Comment:
I did yes; check out the latter half of #2. It's complaining that the parameter passed to the `$callable` needs to be a boolean but we're passing null. Based on the `array_walk` [docs](http://php.net/manual/en/function.array-walk.php), the `$userdata` parameter is optional hence the solution I used here. If there is a better way to solve I'd love to know because it doesn't _seem_ like the cleanest method 🤔

Reply:
A ha! So, you're right — we need to only pass that parameter downstream if the argument is actually provided — but it has nothing to do with `array_walk`.

Specifically, `array_walk` doesn't care what kind of argument is passed there. It's the downstream method that cares. (In this case, `saveElement` is looking for a boolean param, probably to toggle some validation logic. The fact that we were passing a default `null` was fine with `array_walk`, but `saveElement` complained.)

We can't predict whether a target method will ever expect a second param, so we shouldn't presume to put a default value there. We should let our user do that, in cases where they know it's relevant to the target method.

However, a simple truthiness check isn't a good idea here. (What if the `$userdata` is `false`, or `null`, or `[]`, or `0`, or anything else that casts to a false boolean? That argument will never be passed through.) The correct check to use here is `isset()`.